ArmRobot/Assets/Scripts/RandomizerPositionObject.cs

```diff
@@ -0,0 +1,166 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class RandomizerPositionObject : MonoBehaviour
+{
+
+
+    private GameObject gameObjectSeen;
+    float robotMinReach = 0.2f;
+    float robotMaxReach = 0.5f;
+
+    Bounds tableBounds;
+    public float yAltitudeTable = 0.778f;
+
+
+    // CONTROL
+
+    public void Move(List<GameObject> listOfAlreadyMovedObjects)
+    {
+        // random position (on table, within reach)     
+        Vector2 tableTopPoint = RandomReachablePointOnTable(listOfAlreadyMovedObjects);
+        Vector3 tableCenter = tableBounds.center;
+        float x = tableCenter.x + tableTopPoint.x;
+        float z = tableCenter.z + tableTopPoint.y;
+        transform.position = new Vector3(x, yAltitudeTable, z);
+        
+
+        // random rotation
+        Vector3 randomRotation = new Vector3(
+            transform.rotation.eulerAngles.x,
+            Random.value * 360.0f,
+            transform.rotation.eulerAngles.z);
+        transform.rotation = Quaternion.Euler(randomRotation);
+    }
+
+
+    // HELPERS
+
+    Vector2 RandomReachablePointOnTable(List<GameObject> listOfAlreadyMovedObjects)
+    {
+        while (true)
+        {
+            Vector2 randomOffset = RandomPoint(robotMinReach, robotMaxReach, listOfAlreadyMovedObjects);
+            bool onTable = PointOnTable(randomOffset);
+            if (onTable)
+            {
+                return randomOffset;
+            }
+        }
+    }
+
+    bool PointOnTable(Vector2 point)
+    {
+        /*  point: The 2D point on table top, relative to center of table top.
+         *  Determines if this point would be on the table or not.      
+         */
+        GameObject table = GameObject.Find("Table");
+        tableBounds = table.GetComponent<Collider>().bounds;
+        Vector3 tableExtents = tableBounds.extents;
+        float targetRadius = 0f;
+        if (tag == "Sphere"){
+                targetRadius = GameObjectRadius(gameObject); // for the sphere 
+            } 
+        else if (tag == "Cylinder"){
+                targetRadius = GameObjectRadius(gameObject); 
+            }
```

Comment:
The `GameObjectRadius` method already handles all three object types (cube, cylinder, sphere), so you don't need the if/else here. You can replace this all with one `float targetRadius = GameObjectRadius(gameObject);` line.